### ISSUE TITLE
Block size median

### ISSUE THREAD

**Issue by nshopik** [esotericnonsense/bitcoind-ncurses#26]
It would be nice to have block size median for last 144 and 2016 blocks on monitor tab

**Comment by esotericnonsense** [comment]
I recall looking into this in the past. If I remember correctly the issue is that there's no way to do it without querying bitcoind for the last X blocks which can take a lot of time.

I've been musing how to solve these sorts of problems and I can see two reasonable ways:

1) pull requests into Bitcoin Core that add the necessary API calls
2) add some persistent storage (sqlite?) so that the crawling only has to be done once

**Comment by nshopik** [comment]
Is it really slow? On my test machine with 10 year old ` Intel(R) Xeon(TM) CPU 2.40GHz` it takes about 0.025s to query `getblockcount`, `getblockhash` and then `getblock` to get block size. So it takes about 3 seconds to make median of 144 blocks on 10 year old machine. Though getting last 2016 blocks is where it will be noticeable with about 50 seconds to compete.

It maybe good idea to store that somewhere else, to speedup. But I don't think it really helps process unless you are constantly restarting `bitcoind-ncurses, otherwise you already store that in memory.

**Comment by esotericnonsense** [comment]
I think that generally it should aim to use as little CPU as possible and be lightweight.
The original machine I designed this for was an old Intel Atom netbook!

That said, it should be possible to include this in some form. Perhaps on the user's request (so that it doesn't bog down bitcoind with RPC calls needlessly). If we had some way of prioritising RPC calls that would probably work (we don't want getblock calls for this median to block other updates).

WRT restarting it, well, during development I'm restarting it constantly :)

**Comment by nshopik** [comment]
I agree about keep it lightweight as possible. And like idea make it on request, as it seems much simpler solution for now w/o bogging down everything. This could go on separate tab, to differentiate non-live nature of it.
